QualityControl/lib/database/index.js

```diff
@@ -13,19 +13,36 @@
  */
 
 import { LogManager } from '@aliceo2/web-ui';
+import { isRunningInDevelopment, isRunningInProduction, isRunningInTest } from '../utils/environment.js';
 
 const LOG_FACILITY = `${process.env.npm_config_log_label ?? 'qcg'}/database`;
 
 /**
  * Initializes the database connection and runs migrations.
  * @param {object} sequelizeDatabase - The Sequelize database instance.
+ * @param {object} options - Options for database initialization.
+ * @param {boolean} [options.forceSeed=false] - Whether to force seeding the database.
+ * @param {boolean} [options.drop=false] - Whether to drop existing tables before migration (development only).
  * @returns {Promise<void>} A promise that resolves when the database is initialized.
  */
-export const initDatabase = async (sequelizeDatabase) => {
+export const initDatabase = async (sequelizeDatabase, { forceSeed = false, drop = false }) => {
   const _logger = LogManager.getLogger(LOG_FACILITY);
   try {
-    await sequelizeDatabase.connect();
-    await sequelizeDatabase.migrate();
+    if (isRunningInTest) {
+      await sequelizeDatabase.dropAllTables();
+      await sequelizeDatabase.migrate();
+      await sequelizeDatabase.seed();
+    } else if (isRunningInDevelopment) {
+      if (drop) {
+        await sequelizeDatabase.dropAllTables();
+      }
+      await sequelizeDatabase.migrate();
+      if (forceSeed) {
+        await sequelizeDatabase.seed();
+      }
+    } else if (isRunningInProduction) {
+      await sequelizeDatabase.migrate();
+    }
```

Comment:
```suggestion
    } else {
      if (!isRunningInProduction && drop) {
        await sequelizeDatabase.dropAllTables();
      }
      await sequelizeDatabase.migrate();
      if (!isRunningInProduction && forceSeed) {
        await sequelizeDatabase.seed();
      }
    }
```
Its up to you which one you find easier to read.